internal/scheduler/jobdb/job.go

```diff
@@ -413,6 +423,13 @@ func (job *Job) WithPriority(priority uint32) *Job {
 	return j
 }
 
+// WithBidPrice returns a copy of the job with the bidPrice updated.
+func (job *Job) WithBidPrice(price float64) *Job {
+	j := copyJob(*job)
```

Comment:
Arguably you want DeepCopyJob, but then every other With* method has copyJob :(

Reply:
I *think* this is fine!